homeassistant/components/unifi/controller.py

```diff
@@ -0,0 +1,172 @@
+"""UniFi Controller abstraction."""
+
+import asyncio
+import async_timeout
+
+from aiohttp import CookieJar
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST
+from homeassistant.helpers import aiohttp_client
+
+from .const import CONF_CONTROLLER, CONF_POE_CONTROL, LOGGER
+from .errors import AuthenticationRequired, CannotConnect
+
+
+class UniFiController:
+    """Manages a single UniFi Controller."""
+
+    def __init__(self, hass, config_entry):
+        """Initialize the system."""
+        self.hass = hass
+        self.config_entry = config_entry
+        self.available = True
+        self.api = None
+        self.progress = None
+        self._cancel_retry_setup = None
+
+    @property
+    def host(self):
+        """Return the host of this controller."""
+        return self.config_entry.data[CONF_CONTROLLER][CONF_HOST]
+
+    @property
+    def mac(self):
+        """Return the mac address of this controller."""
+        controller_mac = None
+        for client in self.api.clients.values():
+            if self.host == client.ip:
+                controller_mac = client.mac
+                break
```

Comment:
Alternative is to return client.mac here and just return None at the end of function. You also not need to declare controller_mac in that case